Source/JavaScriptCore/wasm/js/WebAssemblyBuiltin.cpp

```diff
@@ -33,6 +33,7 @@
 #include "JSWebAssemblyArrayInlines.h"
 #include "JSWebAssemblyInstance.h"
 #include "JSWebAssemblyRuntimeError.h"
+#include <uchar.h>
```

Comment:
Why is this include needed?

Reply:
> It's needed because this is where char16_t is defined.

I don’t think that’s right. In modern C++ like the C++20 minimum we require, `UChar` may be defined in that header, but `char16_t` is built into the compiler!

Reply:
No, the naming is very confusing but `UChar` (unrelated to uchar.h) is defined by ICU, whereas `char16_t` is defined by the C and C++ standard libraries in uchar.h.

I didn't know this until a few days ago.

Reply:
> No, the naming is very confusing but `UChar` (unrelated to uchar.h) is defined by ICU, whereas `char16_t` is defined by the C and C++ standard libraries in uchar.h.

Yes, I was surprised! Thanks for explaining.

I then researched this and it’s _almost_ right.

In C it’s an integer type in the strings library in `<uchar.h>`, as you say.

https://en.cppreference.com/w/c/string/multibyte/char16_t

In C++, it’s a keyword and does not require a header.

https://en.cppreference.com/w/cpp/keyword/char16_t.html

I think the include should be removed.

Reply:
OK, you're right, good catch. I was looking at https://en.cppreference.com/w/c/string/multibyte/char16_t by mistake. I'll remove the header.